src/key_packages/mod.rs

```diff
@@ -274,22 +275,69 @@ impl KeyPackageBundle {
     pub fn new_with_keypair(
         ciphersuites: &[CiphersuiteName],
         credential_bundle: &CredentialBundle,
-        extensions: Vec<Box<dyn Extension>>,
+        mut extensions: Vec<Box<dyn Extension>>,
         key_pair: HPKEKeyPair,
         leaf_secret: Vec<u8>,
     ) -> Result<Self, ConfigError> {
         debug_assert!(!ciphersuites.is_empty());
```

Comment:
We should return an error `if ciphersuite.is_empty()`.

Reply:
We should, but we might not need `ciphersuites` in the first place. See my reply above.

Reply:
Cann you add this as long as the `ciphersuite` is still there?